test/functional/mempool_accept.py

```diff
@@ -372,5 +374,24 @@ def run_test(self):
             maxfeerate=0,
         )
 
+        self.log.info('Spending a confirmed bare multisig is okay')
+        address = self.wallet.get_address()
+        tx = tx_from_hex(raw_tx_reference)
+        privkey, pubkey = generate_keypair()
+        tx.vout[0].scriptPubKey = keys_to_multisig_script([pubkey] * 3, k=1)  # Some bare multisig script (1-of-3)
+        tx.rehash()
+        self.generateblock(node, address, [tx.serialize().hex()])
+        tx_spend = CTransaction()
+        tx_spend.vin.append(CTxIn(COutPoint(tx.sha256, 0), b""))
+        tx_spend.vout.append(CTxOut(tx.vout[0].nValue - int(fee*COIN), script_to_p2wsh_script(CScript([OP_TRUE]))))
+        tx_spend.rehash()
+        sign_input_legacy(tx_spend, 0, tx.vout[0].scriptPubKey, privkey, sighash_type=SIGHASH_ALL)
+        tx_spend.vin[0].scriptSig = bytes(CScript([OP_0])) + tx_spend.vin[0].scriptSig
```

Comment:
@BrandonOdiwuor For my understanding, why is prepending `bytes(CScript([OP_0]))` to the script sig required after having gone through `sign_input_legacy`?

Reply:
This is due to the `OP_CHECKMULTISG` off-by-one bug. See for example: https://bitcoin.stackexchange.com/questions/116960/why-checkmultisig-bug-cant-be-solved

@instagibbs I think your suggestion here is incorrect? We do need to scriptsig otherwise the tx is simply invalid.

Reply:
Thanks for this clarification @willcl-ark

Reply:
@willcl-ark yeah my comment was nonsense, deleted